src/components/media-loader.js

```diff
@@ -257,13 +259,38 @@ AFRAME.registerComponent("media-loader", {
     }
   },
 
+  refresh() {
+    if (this.networkedEl && !NAF.utils.isMine(this.networkedEl) && !NAF.utils.takeOwnership(this.networkedEl)) return;
+
+    // When we refresh, we bump the version to the current timestamp.
+    //
+    // The only use-case for refresh right now is re-fetching screenshots.
+    this.el.setAttribute("media-loader", { version: Math.floor(Date.now() / 1000) });
+  },
+
   async update(oldData) {
+    const { src, version } = this.data;
+
+    if (!src) return;
+    const srcChanged = oldData.src !== src;
+    const versionChanged = !!(oldData.version && oldData.version !== version);
+
+    if (versionChanged) {
+      this.el.emit("media_refreshing");
+
+      // Don't animate if its a refresh.
+      this.data.animate = false;
```

Comment:
Technically shouldn't be mutating the data object directly like this... but we are so far off the normal aframe path these days not sure if it would be better to do this or setAttribute.

Reply:
oh right. i think i'll leave it, because i'm not sure what the behavior is if we do that within `update()`